x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -369,8 +370,13 @@ export function addNewFeatureToIndex(geometry: Geometry | Position[]) {
     }
 
     try {
-      await (layer as IVectorLayer).addFeature(geometry);
-      await dispatch(syncDataForLayerDueToDrawing(layer));
+      dispatch(updateEditShape(DRAW_SHAPE.WAIT));
+      await asyncForEach(geometries, async (geometry) => {
+        await (layer as IVectorLayer).addFeature(geometry);
+      });
+      await dispatch(syncDataForLayerDueToDrawing(layer)).then(() => {
```

Comment:
Can you move the `.then` content to just be on the next line. Since the method is async and there is an await, `.then` is not needed and makes reading this more difficult and is not in the same style as everything else in this file